components/script/dom/blob.rs

```diff
@@ -222,6 +229,60 @@ impl BlobMethods for Blob {
         let blob_impl = BlobImpl::new_sliced(rel_pos, self.blob_id.clone(), type_string);
         Blob::new(&*self.global(), blob_impl)
     }
+
+    // https://w3c.github.io/FileAPI/#text-method-algo
+    fn Text(&self) -> Rc<Promise> {
+        let in_compartment_proof = AlreadyInCompartment::assert(&self.global());
+        let p = Promise::new_in_current_compartment(
+            &self.global(),
+            InCompartment::Already(&in_compartment_proof),
+        );
+        let id = self.get_blob_url_id();
+        self.global().read_file_async(
+            id,
+            p.clone(),
+            Box::new(|promise, bytes| {
+                let bytes = bytes.unwrap();
```

Comment:
I think both here and in the other method, if `bytes` is an error, you want to reject the promise, instead of panicking.